connect/src/client/hb.js

```diff
@@ -72,14 +72,36 @@ export function requestWith ({ fetch, logger: _logger, HB_URL, signer }) {
         toHttpSigner(signer)(toSigBaseArgs({
           url: joinUrl({ url: HB_URL, path }),
           method,
-          headers,
-          includePath: true
+          headers
+          // includePath: true
         })).then((req) => ({ ...req, body }))
       ))
       .map(logger.tap('Sending HTTP signed message to HB: %o'))
       .chain((request) => of(request)
-        .chain(fromPromise(({ url, method, headers, body }) =>
-          fetch(url, { method, headers, body, redirect: 'follow' })
+        .chain(fromPromise(({ url, method, headers, body }) => {
+          return fetch(url, { method, headers, body, redirect: 'follow' })
+            .then(async res => {
+              if (res.status < 300) {
+                const contentType = res.headers.get('content-type')
+
+                if (contentType && contentType.includes('multipart/form-data')) {
+                  return res
+                } else if (contentType && contentType.includes('application/json')) {
+                  const body = await res.json()
+                  return {
+                    headers,
+                    body
+                  }
+                } else {
+                  const body = await res.text()
+                  return {
+                    headers: res.headers,
+                    body
+                  }
```

Comment:
Any chance that HyperBEAM returns actual bytes? Maybe need a checks for `content-type: application/octet-stream` and then call `res.arrayBuffer()`

Reply:
absolutely, but right now it does not return app/json

Reply:
maybe in future when we have a way to verify